src/scripts/scripts/zone/karazhan/boss_midnight.cpp

```diff
@@ -16,40 +16,46 @@
 
 /* ScriptData
 SDName: Boss_Midnight
-SD%Complete: 90
-SDComment:
+SD%Complete: 95
+SDComment: Should pull the whole room on aggro, Charge Spell & Targeting
 SDCategory: Karazhan
 EndScriptData */
 
 #include "precompiled.h"
 #include "def_karazhan.h"
 
-#define SAY_MIDNIGHT_KILL           -1532000
-#define SAY_APPEAR1                 -1532001
-#define SAY_APPEAR2                 -1532002
-#define SAY_APPEAR3                 -1532003
-#define SAY_MOUNT                   -1532004
-#define SAY_KILL1                   -1532005
-#define SAY_KILL2                   -1532006
-#define SAY_DISARMED                -1532007
-#define SAY_DEATH                   -1532008
-#define SAY_RANDOM1                 -1532009
-#define SAY_RANDOM2                 -1532010
-
-#define SPELL_SHADOWCLEAVE          29832
-#define SPELL_INTANGIBLE_PRESENCE   29833
-#define SPELL_BERSERKER_CHARGE      26561                   //Only when mounted
-#define SPELL_UPPERCUT              29850
-#define SPELL_KNOCKDOWN             29711
-
-#define MOUNTED_DISPLAYID           16040
-
-//Attumen (TODO: Use the summoning spell instead of creature id. It works , but is not convenient for us)
-#define SUMMON_ATTUMEN 15550
+enum MidnightAttumen
+{
+    SAY_MIDNIGHT_KILL           = -1532000,
+    SAY_APPEAR1                 = -1532001,
+    SAY_APPEAR2                 = -1532002,
+    SAY_APPEAR3                 = -1532003,
+    SAY_MOUNT                   = -1532004,
+    SAY_KILL1                   = -1532005,
+    SAY_KILL2                   = -1532006,
+    SAY_DISARMED                = -1532007,
+    SAY_DEATH                   = -1532008,
+    SAY_RANDOM1                 = -1532009,
+    SAY_RANDOM2                 = -1532010,
+
+    SPELL_SHADOWCLEAVE          = 29832,
+    SPELL_INTANGIBLE_PRESENCE   = 29833,
+    SPELL_BERSERKER_CHARGE      = 26561, //Only when mounted
+    SPELL_UPPERCUT              = 29850,
+    SPELL_KNOCKDOWN             = 29711,
+
+    MOUNTED_DISPLAYID           = 16040,
+
+    SPELL_CHARGE_VISUAL         = 40602,
+
+    //Attumen (TODO: Use the summoning spell instead of creature id. It works , but is not convenient for us)
+    SUMMON_ATTUMEN              = 15550
+};
+
```

Comment:
Consider using multiple enums.

An example of mine:

```c++
enum Quotes
{
    YELL_AGGRO    = -1564000, // You will die in the name of Lady Vashj!
    ...
};

enum Spells
{
    SPELL_CREATE_NAJENTUS_SPINE   = 39956,
    ...

};

enum GameObjects
{
    GOBJECT_SPINE = 185584
};
```
https://github.com/Looking4Group/L4G_Core/pull/3084/files#diff-34fe603c2b18d6f32692473b3d7a5f83L27